google/cloud/bigtable/table.cc

```diff
@@ -171,6 +172,7 @@ std::vector<FailedMutation> Table::BulkApply(BulkMutation mut) {
     if (!status.ok() && !retry_policy->OnFailure(status)) {
       break;
     }
+    if (!mutator.HasPendingMutations()) break;
     auto delay = backoff_policy->OnCompletion(status);
     std::this_thread::sleep_for(delay);
```

Comment:
Why do we backoff on success at all?

Reply:
It's just how the RPC works. The response itself has fields for errors. So the overall response can be OK, but still point us to failed mutations that we should retry: https://github.com/googleapis/googleapis/blob/1848f0849e4c18168e39c796127c705eaa03b776/google/bigtable/v2/bigtable.proto#L411-L427

Reply:
Because this API can be partially successful.  The RPC returns successfully with a list of mutations that failed with a transient error.

Reply:
Thanks.